libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -0,0 +1,418 @@
+    ///
+    /// Assigning Unity components to the <see cref="Source"/> and <see cref="Receiver"/> fields serves
+    /// as an indication of the user intent to send and/or receive media through the transceiver, and is
+    /// used during the SDP exchange to derive the <see xref="WebRTC.Transceiver.Direction"/> to negotiate.
+    /// After the SDP negotiation is completed, the <see cref="Transceiver"/> property refers to the transceiver
+    /// associated with this media line, and which the sender and receiver will use.
+    ///
+    /// Users typically interact with this class through the peer connection transceiver collection in the Unity
+    /// inspector window, though direct manipulation via code is also possible.
+    /// </summary>
+    [Serializable]
+    public class MediaLine
+    {
+        /// <summary>
+        /// Backing field to serialize the <see cref="Kind"/> property.
+        /// </summary>
+        /// <seealso cref="Kind"/>
+        [SerializeField]
+        private MediaKind _kind;
+
+        /// <summary>
+        /// Kind of media of the media line and its attached transceiver.
+        ///
+        /// This is assiged when the media line is created with <see cref="PeerConnection.AddMediaLine(MediaKind)"/>
+        /// and is immutable for the lifetime of the peer connection.
+        /// </summary>
+        public MediaKind Kind => _kind;
+
+        /// <summary>
+        /// Backing field to serialize the <see cref="Source"/> property.
+        /// </summary>
+        /// <seealso cref="Source"/>
+        [SerializeField]
+        private MediaTrackSource _source;
+
+        /// <summary>
+        /// Media source producing the media to send through the transceiver attached to this media line.
+        /// This must be an instance of a class derived from <see cref="AudioTrackSource"/> or <see cref="VideoTrackSource"/>
+        /// depending on whether <see cref="Kind"/> is <see xref="Microsoft.MixedReality.WebRTC.MediaKind.Audio"/>
+        /// or <see xref="Microsoft.MixedReality.WebRTC.MediaKind.Video"/>, respectively.
+        /// 
+        /// Internally the peer connection will automatically create and manage a media track to bridge the
+        /// media source with the transceiver.
+        ///
+        /// If this is non-<c>null</c> then the peer connection will negotiate sending some media, otherwise
+        /// it will signal the remote peer that it does not wish to send (receive-only or inactive).
+        ///
+        /// If <see cref="Transceiver"/> is valid, that is a first session negotiation has already been completed,
+        /// then changing this value raises a <see xref="WebRTC.PeerConnection.RenegotiationNeeded"/> event on the
+        /// peer connection of <see cref="Transceiver"/>.
+        /// </summary>
+        public MediaTrackSource Source
+        {
+            get { return _source; }
+            set
+            {
+                if (_source != value)
```

Comment:
There is a cast in `UpdateSenderOnSourceChanged`, but should we check the value type for correctness explicitly here for clarity?

Reply:
Fixed; the setter now correctly checks the type of `value`.